sycl/source/detail/config.def

```diff
@@ -26,7 +26,7 @@ CONFIG(SYCL_PROGRAM_APPEND_COMPILE_OPTIONS, 64, __SYCL_PROGRAM_APPEND_COMPILE_OP
 CONFIG(SYCL_HOST_UNIFIED_MEMORY, 1, __SYCL_HOST_UNIFIED_MEMORY)
 // 260 (Windows limit) - 12 (filename) - 84 (cache directory structure)
 CONFIG(SYCL_CACHE_DIR, 164, __SYCL_CACHE_DIR)
-CONFIG(SYCL_CACHE_TRACE, 1, __SYCL_CACHE_TRACE)
+CONFIG(SYCL_CACHE_TRACE, 16, __SYCL_CACHE_TRACE)
```

Comment:
what is this, anyway? Doesn't it set the default values for these environment vars? But, if so, why was it set to 1 before?

Reply:
That's a good question. From what I understand, the syntax is `CONFIG(CONFIG_NAME, VALUE_MAX_SIZE_IN_BYTES, COMPILE_TIME_CONFIG_NAME)` and `VALUE_MAX_SIZE_IN_BYTES` is the maximum amount of bytes to read when we read this environment variable from the configuration file. 

I do not think there's a default - if I remove this line, I get  the following error:
```
llvm/sycl/source/detail/config.hpp:711:30: error: ‘SYCL_CACHE_TRACE’ was not declared in this scope; did you mean ‘SYCL_UR_TRACE’?
  711 | template <> class SYCLConfig<SYCL_CACHE_TRACE> {
```

Now, regarding the size, I thing it was `1` earlier because this flag was just accepting a boolean - to disable/enable tracing of disk cache. Now, it is accepting an integer so we need to updated the size accordingly. I used `16` bytes previously because I was copying what `SYCL_UR_TRACE` did. But, now that I think more, I guess `4` bytes should be sufficient.